topic #4 Generic vs Convenience approach for Spatial data APIs/README.md

```diff
@@ -1,2 +1,281 @@
+
+Setting up the engine consists of three steps:
+- The engine itself has to be setup. This provides the landingpage, conformance page and the API page.
+- The Features part has to be activated. This will result in the engine responding to the collections page.
+- The API has to be rebuild and the URL routing has to be setup.
+
+The datasource consists of providing an implementation of five methods:
+- Two methods concerned with collections. These are the method that will return all the collections and the method returning information about a specific collection.
+- Two methods concerned with features/ items. One method will return a list of features, the other one will return a single feature.
+- One method that configures the OpenAPI (Swagger) specification.
+
+### Usage example
+
+The following code fragment shows the minimum requirements for creating the engine and mounting it inside an existing router.
+
+```go
+func EnableOGCAPIFeatures() {
+  engine := apif.NewSimpleEngine(mountingPath)
+  apif.AddBaseJSONTemplates(engine)
+  apif.AddBaseHTMLTemplates(engine)
+
+  featuredatasource := myDatasourceImplementation.NewImplementation()
+  apif.EnableFeatures(engine, featuredatasource)
+  apif.AddFeaturesJSONTemplates(engine)
+  apif.AddFeaturesHTMLTemplates(engine)
+
+  engine.RebuildOpenAPI()
+  router.HandleFunc(regexp.MustCompile("^"+mountingPath), engine.HTTPHandler)
+}
+```
+
+In order to provide a datasource implementation for features, the following methods must be implemented:
+
+```go
+type FeatureService interface {
+  Collections() []featuremodels.Collection
+  Collection(string) featuremodels.Collection
+  Features(*http.Request, *featuremodels.FeaturesParams) featuremodels.Features
+  Feature(string, string) *featuremodels.Feature
+  BuildOpenAPISpecification(builder coreservices.OpenAPIBuilder)
+}
+```
+
+The readme of the library contains a code fragment that implements the datasource. The readme can be found at https://github.com/purple-polar-bear/go-ogc-api
+
+## Results
+This chapter contains the results from our research.
+
+### Library
+Before we dive into the benefits and the cost of an additional OGC API Features implementation, we will look into the results of the library development.
+
+#### Code review on library
+
+The design of the library as well as the implementation have been submitted to the project owners of Goaf (a team of Kadaster). They were asked to criticize the design. They find the implementation, at face value, sufficient for further exploration. They want to use this library for a different project. The have also stated that a conclusion cannot be drawn without actual using the code in a different project.
```

Comment:
"They want to use this libray for a different project" <- niet geheel juist
- we zitten in de toekomst met meerdere OGC API gerelateerde projecten
- m.b.t deze library, klopt "sufficient for further exploration" dus eerst (los van code review enz..) PoC implementatie voor dat dingen richting PROD o.i.d. gaan

Reply:
OK, tekst verhelderd. Is dit ok?